### PR TITLE
planner: query should report error when timestamp overflow

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: yisaer <disxiaofei@163.com>

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/27242

Problem Summary:

https://github.com/pingcap/tidb/pull/26584 introduce forbidding using timestamp which out of valid range, that causes tidb-server build dual table plan if the query condition contains invalid timstamp.

### What is changed and how it works?

A proper behavior is to forbid query and report error I think.

### Check List 

Tests 

- [x] Unit test

### Release note 

```release-note
Fix the error that query won't report error when timestamp overflow in query condition.
```

### CODE REVIEW

File: util/ranger/detacher.go [hunk 1]
```diff
@@ -617,7 +617,7 @@ func (d *rangeDetacher) detachDNFCondAndBuildRangeForIndex(condition *expression
 			var accesses, filters []expression.Expression
 			res, err := d.detachCNFCondAndBuildRangeForIndex(cnfItems, newTpSlice, true)
 			if err != nil {
-				return nil, nil, nil, false, nil
+				return nil, nil, nil, false, errors.Trace(err)
```
Comment (wjhuang2016) [inline util/ranger/detacher.go:620]: Returning `err` is enough.

Comment (Yisaer) [reply util/ranger/detacher.go]: updated.

**Review by wjhuang2016 — APPROVED** [review]

**Review by rebelice — APPROVED** [review]

**Pull request merged**